swap_meet/vendor.py

```diff
@@ -1,2 +1,91 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if not inventory:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+
+    def get_by_id(self, item_id):
+        for item in self.inventory:
+            if item.id == item_id:
+                return item
+        
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        other_vendor.inventory.remove(their_item)
+        self.inventory.append(their_item)
+        self.inventory.remove(my_item)
+        other_vendor.inventory.append(my_item)
+        return True
+    
+    def swap_first_item(self, other_vendor):
+        if not self.inventory or not other_vendor.inventory:
+            return False
+        first_item = self.inventory[0]
+        first_item_other = other_vendor.inventory[0]
+        if  first_item  and first_item_other:
+            self.inventory.remove(first_item)
+            other_vendor.inventory.append(first_item)
+            other_vendor.inventory.remove(first_item_other)
+            self.inventory.append(first_item_other)
+            return True
+    
+    def get_by_category(self, category="Unknown"):
+        return[item for item in self.inventory if item.get_category() == category]
+    
+    def get_highest_item(self, list_items, key=lambda x: x):
+        if not list_items:
+            return False
+        
+        highest = list_items[0]
+        for item in list_items:
+            if key(item) > key(highest):
+                highest = item
+        return highest
+    
+    def get_best_by_category(self, category):
+        items_in_category = self.get_by_category(category)
+
+        if not items_in_category:
+            return False
```

Comment:
What should this method return if there are no items in the inventory that match the category? Review the README to double check.

Reply:
✅ Fixed (will push later!)